src/Context.h

```diff
@@ -84,8 +84,8 @@ class Context
 public:
   CLI2                                cli2                {};
   std::string                         home_dir            {};
-  File                                rc_file             {"~/.taskrc"};
-  Path                                data_dir            {"~/.task"};
+  File                                rc_file             {"~/.config/task/taskrc"};
+  Path                                data_dir            {"~/.local/share/task"};
```

Comment:
@sl4mmy

3) Do NOT even think about handling my data or config in any case. Let alone sneakily.

Warn me, haunt me, persuade me. But DO NOT TOUCH my data or config files.

Reply:
In my personal experience, every time I've seen some auto-magic like in step 3, it's led to problems of some sort. Also, I'm wondering if we need step 1.

As a result, what do you think about this plan?
1. Make `XDG_*` the defaults, keep supporting the old locations
2. Add a warning to a future 2.x release that warns the old location won't be supported anymore in upcoming 3.0
3. When 3.0 comes out, drop support for the old locations but leave the warning from step 2 in place. until say 3.1

Reply:
Stop bothering about dropping support for the current locations.

Just use the new locations  for vanilla setups.

Support old locations indefinitely w/ precedence of old locations, warn if doing so and new locations exist. Stay quiet in case no new locations hold configs or data. Usr knows better then. Docs reflect this clearly and boldly.

Reply:
Done, see latest changes :-)